src/cronjobs.js

```diff
@@ -47,6 +49,318 @@ const start = () => {
+                                                Array.isArray(
+                                                    campaign.allocations
+                                                )
+                                            ) {
+                                                if (bigquery_campaign_id) {
+                                                    // fiter adsets only with bigquery id
+                                                    campaign.allocations =
+                                                        campaign.allocations.filter(
+                                                            adset =>
+                                                                adset.bigquery_adset_id
+                                                        );
+
+                                                    // adsets
+                                                    for (const adset of campaign.allocations) {
+                                                        const {
+                                                            // budget: adsetBudget,
+                                                            bigquery_adset_id,
+                                                        } = adset;
+
+                                                        if (bigquery_adset_id) {
+                                                            // get spending by period
+                                                            const spending = (
+                                                                await getBigquerySpending(
+                                                                    {
+                                                                        campaignId:
+                                                                            bigquery_campaign_id,
+                                                                        adsetId:
+                                                                            bigquery_adset_id,
+                                                                        period: period.label,
+                                                                    }
+                                                                )
+                                                            )[0];
+
+                                                            const adsetMetrics =
+                                                                getMetrics({
+                                                                    period: period.label,
+                                                                    periodBudget:
+                                                                        campaignBudget,
+                                                                    spending,
+                                                                });
+
+                                                            adset.mtd_spent =
```

Comment:
@PedroLabrador this logic seem to be repeated in multiples by instance  in line 212 is the same logic with different variables same on line 249 and 298 
it's seem like that piece of code is repeated we should move that into a abstraction a function this patter or rule in programming is know as  DRY (Don't Repeat Your Self)